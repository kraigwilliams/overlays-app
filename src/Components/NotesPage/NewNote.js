import React,{Component} from 'react';
import config from '../../config'
import TokenService from '../../services/token-service'

class newNote extends Component{

    state={
       topics:[]
    }

    componentDidMount(){
      document.title ="Enter New Note - Overlays"
         fetch(`${config.API_ENDPOINT}/topics`,{
        headers:{
          authorization:"bearer " + TokenService.getAuthToken()
        }
      })
      .then(res =>res.json()
      .then(data=>{
        
        this.setState({topics:data})})
      ) 



    }
onSubmit=event=>{
    event.preventDefault()
    const {note_title,note_contents,from_topic}= event.target
    console.log("trying to get article id",from_topic.value)
    fetch(`${config.API_ENDPOINT}/notes`,{
    
        method: "POST",
        headers: {
          "content-type": "application/json",
          authorization:"bearer " + TokenService.getAuthToken()
        },
        body: JSON.stringify({
          note_title: note_title.value,
          note_contents: note_contents.value,
          from_topic:from_topic.value
          
          
        })
      })
        .then(res => res.json())
        .then( 
          // this.componentDidMount()
          this.props.history.push('/topics')
          //console.log(data);
        );
}


render(){

    return(
        <>
        <h3>Make a new Note</h3>
        <form className="note-form" onSubmit={this.onSubmit}>
          <label id="note-label" htmlFor="note_title">Enter Note Title</label>
            <input type="text" id="note_title" name="note_title"placeholder="Note Title"/><br/><br/>
            <label id="note-label" htmlFor="note_contents">Enter Note Content</label>
            <textarea id="note_contents" name="note_contents" placeholder="Enter Note Text"></textarea><br/><br/>
            <select name="from_topic">
                {this.state.topics.map(topic=>{
                    return <option name={topic.id} value={topic.id}key={topic.id}>{topic.topic_name}</option>
                })}
            </select>
            <input type="submit"/>


        </form>
        </>
    )
}


}


export default newNote