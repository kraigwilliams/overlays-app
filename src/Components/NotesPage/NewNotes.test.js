import React from 'react';
//import { render } from '@testing-library/react';
import NewNote from './NewNote'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<NewNote/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><NewNote/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><NewNote /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})