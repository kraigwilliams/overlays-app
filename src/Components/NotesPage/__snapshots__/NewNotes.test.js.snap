// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewNote/> renders the UI as expected 1`] = `
Array [
  <h3>
    Make a new Note
  </h3>,
  <form
    className="note-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="note_title"
      id="note-label"
    >
      Enter Note Title
    </label>
    <input
      className="Input "
      id="note_title"
      name="note_title"
      placeholder="Note Title"
      required={true}
      type="text"
    />
    <br />
    <br />
    <label
      htmlFor="note_contents"
      id="note-label"
    >
      Enter Note Content
    </label>
    <textarea
      className="Textarea "
      id="note_contents"
      name="note_contents"
      placeholder="Enter Note Text"
    />
    <br />
    <br />
    <select
      className="Select note-select"
      name="from_topic"
    />
    <button
      className="Button submit-topic-btn"
      type="submit"
    >
      Submit
    </button>
  </form>,
]
`;
